File: desafio-04/lrlucena/scala/xadrez.scala
object Xadrez extends App {
  val peças = List("", "Peão", "Bispo", "Cavalo", "Torre", "Rainha", "Rei")

  def readLines(n: Int) = (1 to n).map(_ => io.StdIn.readLine())

  // Ler 8 linhas e juntar em um texto, depois transformar em números
  val tabuleiro = readLines(8).mkString.map(_ - '0')

  // Varrer o tabuleiro e selecionar as peças de cada tipo, depois contar
  for(i <- 1 to 6) {
    val num = tabuleiro.filter(_ == i).length
    println(s"${peças(i)}: ${num} peças")
  }
}

File: desafio-04/laurosn/python/xadrez.py
#!/usr/bin/python
import numpy as np

def main():
    f = open('xadrez.txt', 'r')
    l = [[int(num) for num in line.split(' ')] for line in f]
    y = np.array(l)
    print("Peão:" + str(np.count_nonzero(y == 1)) + " peça(s)")
    print("Bispo:" + str(np.count_nonzero(y == 2)) + " peça(s)")
    print("Cavalo:" + str(np.count_nonzero(y == 3)) + " peça(s)")
    print("Torre:" + str(np.count_nonzero(y == 4)) + " peça(s)")
    print("Rainha:" + str(np.count_nonzero(y == 5)) + " peça(s)")
    print("Rei:" + str(np.count_nonzero(y == 6)) + " peça(s)")
if __name__ == "__main__":
  main()